includes/Core/Util/Input.php

```diff
@@ -18,6 +18,26 @@
  * @ignore
  */
 class Input {
+	/**
+	 * Map of input type to superglobal array.
+	 *
+	 * For use as fallback only.
+	 *
+	 * @var array
+	 */
+	protected $map;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function __construct() {
+		$this->map = array(
```

Comment:
Maybe we should add a comment above this hinting at the bug with `INPUT_ENV` and `INPUT_SERVER` (to explain why only those are here).